pkg/controller/gitopsservice/argocd.go

```diff
@@ -0,0 +1,45 @@
+package gitopsservice
+
+import (
+	argoapp "github.com/argoproj-labs/argocd-operator/pkg/apis/argoproj/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"gopkg.in/yaml.v2"
```

Comment:
This should probably use `sigs.k8s.io/yaml` which has some subtle changes _on top_ of go yaml.